receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -254,7 +254,8 @@ func (c *tracesConsumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSe
 		session.MarkMessage(message, "")
 
 		ctx := obsreport.ReceiverContext(session.Context(), c.id, transport)
-		ctx = obsreport.StartTraceDataReceiveOp(ctx, c.id, transport)
+		rec := obsreport.NewReceiver(obsreport.ReceiverSettings{ReceiverID: c.id, Transport: transport})
```

Comment:
Same here, you can cache to object. Please apply the caching to all the non-test code where you can.

Reply:
Ok I applied to caching to all non-test code in my latest commit